app/controllers/posts_controller.rb

```diff
@@ -10,4 +10,19 @@ def show
     @post = Post.find(params[:id])
     @users = User.all.order(id: :asc)
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(params.require(:post).permit(:title, :text))
+    @post.author = current_user
+
+    if @post.save
+      redirect_to user_path(current_user.id)
+    else
+      render :new
+    end
+  end
```

Comment:
[OPTIONAL]
- It would be awesome if you privately added a method for the post parameters and passed them in like 
`@post = @current_user.posts.new(post_params)`
Instead of that in line 19. The method can look like this
```
    private 
    def post_params
      params.require(:post).permit(:title, :text)
    end
```
- You can read more bout this [here](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides)